ispyb/sp/xtalimaging.py

```diff
@@ -49,6 +49,38 @@ def upsert_sample_image_auto_score(
             args=[image_full_path, schema_name, score_class, probability],
         )
 
+    def insert_subsample_for_image_full_path(
+        self,
+        image_full_path,
+        source,
+        position1x,
+        position1y,
+        position2x=None,
+        position2y=None,
+    ):
+        """Store new subsample for a given sample image.
+
+        :param image_full_path: The full path to the sample image
+        :param source: manual or auto
+        :param position1x: x component of position, or if position2x,y given, then x component of top left corner of ROI box
+        :param position1y: y component of position, or if position2x,y given, then y component of top left corner of ROI box
+        :param position2x: x component of lower right corner of ROI box
+        :param position2y: y component of lower right corner of ROI box
```

Comment:
Can you include type information in the docstring? Personally I prefer ["google style" docstrings](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/index.html#google-vs-numpy) but I see what you are using is consistent with the rest of the file. E.g. it isn't clear to me whether the positions should be integers or floats. Also, what units are the positions in?

For this style docstring I believe it would be e.g.:
```
:param image_full_path: The full path to the sample image
:type image_full_path: str
```

Reply:
It might also be worth adding a line of explanation to the docstring that you can either specify a point (by providing `position1x` and `position1y`) or a ROI box (by additionally providing `position2x` and `position2y`). I had to read the parameter descriptions several times before I understood what they meant.

Reply:
Possibly also line breaks in the docstring to avoid too long a line length? E.g. [black by default uses 88 characters](https://black.readthedocs.io/en/stable/the_black_code_style.html#line-length).

Reply:
Thank you, I've done all of this now.